Cant compile for an arm

Consolidate compiler generated dependencies of target utils
[  8%] Building CXX object CMakeFiles/utils.dir/utils.cpp.o
clang++: error: the clang compiler does not support '-mcpu=native'
make[2]: *** [CMakeFiles/utils.dir/build.make:76: CMakeFiles/utils.dir/utils.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:110: CMakeFiles/utils.dir/all] Error 2
make: *** [Makefile:101: all] Error 2

if i will intentionally delete all -mcpu=native flags, then

[  8%] Built target utils
Consolidate compiler generated dependencies of target ggml
[  8%] Built target ggml
[  8%] Building CXX object CMakeFiles/llama.dir/llama.cpp.o
/home/gh228df/llama.cpp/llama.cpp:1447:33: warning: cast from 'const char *' to 'char *' drops const qualifier [-Wcast-qual]
            finp.read ((char *) word.data(), len);
                                ^
/home/gh228df/llama.cpp/llama.cpp:1448:33: warning: cast from 'const char *' to 'char *' drops const qualifier [-Wcast-qual]
            fout.write((char *) word.data(), len);
                                ^
2 warnings generated.
[  8%] Linking CXX static library libllama.a
[  8%] Built target llama
[  8%] Building CXX object CMakeFiles/main.dir/main.cpp.o
/home/gh228df/llama.cpp/main.cpp:356:20: warning: unused variable 'embeddings' [-Wunused-variable]
        const auto embeddings = llama_get_embeddings(ctx);
                   ^
/home/gh228df/llama.cpp/main.cpp:481:28: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]
                    printf(buffer.c_str());
                           ^~~~~~~~~~~~~~
/home/gh228df/llama.cpp/main.cpp:481:28: note: treat the string as an argument to avoid this
                    printf(buffer.c_str());
                           ^
                           "%s", 
1 warning and 1 error generated.
make[2]: *** [CMakeFiles/main.dir/build.make:76: CMakeFiles/main.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:191: CMakeFiles/main.dir/all] Error 2
make: *** [Makefile:101: all] Error 2

I have the same issue when I try to build for Android from Manjaro Linux using the [instructions](https://github.com/ggerganov/llama.cpp#android) in the README. I am on commit `5a5f8b1501fbb34367225544010ddfc306d6d2fe` (latest as of right now). The full error message:

```
.../llama.cpp/build-android (master)> make                                                                                                                                                                                (llama.cpp) 
[  6%] Building C object CMakeFiles/ggml.dir/ggml.c.o
clang: error: the clang compiler does not support '-mcpu=native'
make[2]: *** [CMakeFiles/ggml.dir/build.make:76: CMakeFiles/ggml.dir/ggml.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:193: CMakeFiles/ggml.dir/all] Error 2
make: *** [Makefile:101: all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -386,11 +386,6 @@ if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES
     if (MSVC)
         # TODO: arm msvc?
     else()
-        if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64")
-            # Apple M1, M2, etc.
-            # Raspberry Pi 3, 4, Zero 2 (64-bit)
-            add_compile_options(-mcpu=native)
-        endif()
         if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "armv6")
             # Raspberry Pi 1, Zero
             add_compile_options(-mfpu=neon-fp-armv8 -mfp16-format=ieee -mno-unaligned-access)

